ffi/diplomat/c/examples/segmenter_tiny/test.c

```diff
@@ -0,0 +1,42 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#include "../../include/ICU4XLineSegmenter.h"
+#include <string.h>
+#include <stdio.h>
+
+int main(int argc, char *argv[]) {
+    ICU4XDataProvider* provider = ICU4XDataProvider_create_compiled();
+
+    diplomat_result_box_ICU4XLineSegmenter_ICU4XError segmenter_result = ICU4XLineSegmenter_create_auto(provider);
+    if (!segmenter_result.is_ok)  {
```

Comment:
`ICU4XLineSegmenter_create_auto` is infallible for compiled data, so you *could* do some shenanigans with that.

Reply:
Acknowledged; do you want me to do that here or can I just leave this as-is until 2.0 since we don't have infallible constructors in FFI

Reply:
Don't care